creating relation between Groups, InstallmentPurchase and RecurringBill

wfbraga opened the pull request:
As per Trello Task:

https://trello.com/c/wc7a41FH/8-criar-relacionamento-entre-group-e-installmentpurchase-recurringbill

Factory Creation for InstallmentPurchase
Test refactoring for InstallmentPurchase
Minor improvements on Group model and fatory

wfbraga committed:
creating relation between Groups, InstallmentPurchase and RecurringBill as per Trello Task:
https://trello.com/c/wc7a41FH/8-criar-relacionamento-entre-group-e-installmentpurchase-recurringbill

Factory Creation for InstallmentPurchase
Test refactoring for InstallmentPurchase
Minor improvements on Group model and fatory

brunoao86 commented on spec/factories/installment_purchase.rb:
```diff
@@ -0,0 +1,19 @@
+
+  factory :installment_purchase do
+    description { Faker::Company.name }
+    total_amount { rand(100..100_000) }
```
We need to have the same value here and at the `installment_amount` math:

```suggestion
    total_amount { rand_total_amount }
```

brunoao86 commented on spec/factories/installment_purchase.rb:
```diff
@@ -0,0 +1,19 @@
+    installment_amount { rand_total_amount / rand_installments_quantity }
+    current_installment { rand(1..rand_installments_quantity) }
+    due_day { rand(1..20) }
+    status { %i[active on_hold deactivated].sample }
```
Maybe here it'd be better to have the `active` status as default and [traits](https://github.com/thoughtbot/factory_bot/blob/main/GETTING_STARTED.md#traits) to allow creation of other statuses when needed.

Example:

```ruby
create(:installment_purchase) # active status
create(:installment_purchase, :on_hold)
create(:installment_purchase, :deactivated)
```

wfbraga committed:
Creating 'join' model UserGroup as per Trello task (#9)

* Creating 'join' model UserGroup as per Trello task:
https://trello.com/c/2NKp6Zfa/9-criar-modelo-usergroup

Updating models User and Group to correspond the join relation throuth UserGroup
Updating and inproving tests for User and Group models to test the new relation
Creating factory and tests for UserGroups
Updating seed
